backends-velox/src/main/scala/org/apache/gluten/execution/GenerateExecTransformer.scala

```diff
@@ -243,7 +243,11 @@ object PullOutGenerateProjectHelper extends PullOutProjectHelper {
               case jsonPath if jsonPath.foldable =>
                 Option(jsonPath.eval()) match {
                   case Some(path) =>
-                    GetJsonObject(jsonObj, Literal.create(JSON_PATH_PREFIX + path))
+                    if (path.toString.contains('.')) {
+                      GetJsonObject(jsonObj, Literal.create("$[" + path + "]"))
+                    } else {
+                      GetJsonObject(jsonObj, Literal.create(JSON_PATH_PREFIX + path.toString))
```

Comment:
I just noted that for `json_tuple`, the JSON path is always used for getting top level data, not for accessing data in nested JSON structures.

Can we uniformly convert path to `"$[" + path + "]"` even though it doesn't contain a dot? Then, we can also fix this issue for non-foldable jsonPath by applying this conversion at line 258.

Reply:
fixed,thanks~